include/pybind11/numpy.h

```diff
@@ -328,8 +328,9 @@ class array : public buffer {
 
     array() : array(0, static_cast<const double *>(nullptr)) {}
 
-    array(const pybind11::dtype &dt, const std::vector<size_t> &shape,
-          const std::vector<size_t> &strides, const void *ptr = nullptr,
+    template <typename ST, typename DT>
```

Comment:
Maybe `s/ST/Shape/`, `s/DT/Strides/` (in other overloads too) so the signatures are a bit less cryptic?